src/seedu/addressbook/Main.java

```diff
@@ -52,6 +52,7 @@ private void start(String[] launchArgs) {
         try {
             this.ui = new TextUi();
             this.storage = initializeStorage(launchArgs);
+            this.storage.passUI(ui);
```

Comment:
by doing this, you are unnecessarily creating a dependency between the storage and UI. as you will learn in later lectures, this is not a good practise